public/app/features/explore/ExploreGraphNGPanel.tsx

```diff
@@ -0,0 +1,180 @@
+  GrafanaTheme,
+  TimeZone,
+  TIME_SERIES_VALUE_FIELD_NAME,
+} from '@grafana/data';
+import {
+  Collapse,
+  DrawStyle,
+  GraphNG,
+  GraphNGLegendEvent,
+  Icon,
+  LegendDisplayMode,
+  TooltipPlugin,
+  useStyles,
+  useTheme,
+  ZoomPlugin,
+} from '@grafana/ui';
+import { hideSeriesConfigFactory } from 'app/plugins/panel/timeseries/hideSeriesConfigFactory';
+import { ContextMenuPlugin } from 'app/plugins/panel/timeseries/plugins/ContextMenuPlugin';
+import { ExemplarsPlugin } from 'app/plugins/panel/timeseries/plugins/ExemplarsPlugin';
+import { css, cx } from 'emotion';
+import React, { useCallback, useEffect, useState } from 'react';
+import { splitOpen } from './state/main';
+import { getFieldLinksForExplore } from './utils/links';
+
+const MAX_NUMBER_OF_TIME_SERIES = 20;
+
+interface Props {
+  data: DataFrame[];
+  annotations?: DataFrame[];
+  isLoading: boolean;
+  width: number;
+  absoluteRange: AbsoluteTimeRange;
+  timeZone: TimeZone;
+  onUpdateTimeRange: (absoluteRange: AbsoluteTimeRange) => void;
+  splitOpenFn: typeof splitOpen;
+}
+
+export function ExploreGraphNGPanel({
+  width,
+  data,
+  timeZone,
+  absoluteRange,
+  onUpdateTimeRange,
+  isLoading,
+  annotations,
+  splitOpenFn,
+}: Props) {
+  const [showAllTimeSeries, setShowAllTimeSeries] = useState(false);
+  const theme = useTheme();
+  const [dataFramesWithConfig, setDataFramesWithConfig] = useState<DataFrame[]>();
+  const style = useStyles(getStyles);
+  const timeRange = {
+    from: dateTime(absoluteRange.from),
+    to: dateTime(absoluteRange.to),
+    raw: {
+      from: dateTime(absoluteRange.from),
+      to: dateTime(absoluteRange.to),
+    },
+  };
+
+  const onLegendClick = useCallback(
+    (event: GraphNGLegendEvent) => {
+      const configWithOverrides = hideSeriesConfigFactory(event, { defaults: fieldConfig, overrides: [] }, data);
+      const applied = applyFieldOverrides({
+        fieldConfig: configWithOverrides,
+        data,
+        timeZone,
+        replaceVariables: value => value, // TODO: replace with real replace
+        theme,
+      });
+      setDataFramesWithConfig(applied);
+    },
+    [fieldConfig, data, timeZone, theme]
+  );
+
+  useEffect(() => {
+    setDataFramesWithConfig(addConfigToDataFrame(data));
+  }, [data]);
+
+  if (!dataFramesWithConfig) {
+    return null;
+  }
+
+  const seriesToShow = showAllTimeSeries
+    ? dataFramesWithConfig
+    : dataFramesWithConfig.slice(0, MAX_NUMBER_OF_TIME_SERIES);
+
+  const getFieldLinks = (field: Field, rowIndex: number) => {
+    return getFieldLinksForExplore({ field, rowIndex, splitOpenFn, range: timeRange });
+  };
+
+  return (
+    <>
+      {dataFramesWithConfig.length > MAX_NUMBER_OF_TIME_SERIES && !showAllTimeSeries && (
+        <div className={cx([style.timeSeriesDisclaimer])}>
+          <Icon className={style.disclaimerIcon} name="exclamation-triangle" />
+          {`Showing only ${MAX_NUMBER_OF_TIME_SERIES} time series. `}
+          <span
```

Comment:
Not sure. This is how it was in the old graph component. Should we change this to a button? @jessover9000 
![Screenshot 2021-01-13 at 13 33 11](https://user-images.githubusercontent.com/13729989/104452874-e5e5b980-55a3-11eb-8253-7d638ff444f2.png)

Reply:
I lack a bit of context here. Where does this notification box appear exactly and what do users have to do to trigger it?

Reply:
So this is in Explore shown when we have more than 20 series in the graph.

Reply:
But this is how it currently looks like in the current graph panel as well. So no change here, right?

Reply:
I was only thinking about the semantics of the element rather than its style given that, in fact, it acts like a button.
But it's definitely not blocking given also that this is what we already have